### PR TITLE
Sambev/remove more rules

### PR DESCRIPTION
**Pull request by sambev**
This MR heavily edits the base rules:

- Adds the [eslint recommended rules](https://github.com/eslint/eslint/blob/main/conf/eslint-recommended.js). These are all super solid rules that don't have anything to do with styling and focus on avoiding deprecated or outright bad coding practices.
- Adds the [react/jsx recommended rules](https://github.com/jsx-eslint/eslint-plugin-react#list-of-supported-rules). These are also pretty solid recommendations. We have a few rules on top and some customizations. I'm still pretty open to changing these
- Removes a ton of rules that are already taken care of by prettier
- Removes a bunch of rules I think are unnecessary 🙃.
- Removes some node specific and deprecated rules.
- Note that many removals are also because they are redundant with the recommended settings.

The idea I'm going for is this project is:
- Fix a bunch of overlooked rules
- Focus on frontend react projects
- Focus on creating the smallest rule set possible. This lets teams do fewer overrides, which are probably generally bad (with some exceptions). This also lets teams make additions, which are fine as long as the teams are on board.
- Focus on objectively bad coding practices, not style or formatting

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ module.exports = {
     node: true,
     es6: true
   },
+  extends: ["eslint:recommended", "plugin:react/recommended", "plugin:react-hooks/recommended"],
```
Comment (whiteinge) [inline index.js:8]: We don't currently have react-hooks as a peer dep. Does this fail silently or should we add that dep explicitly?

Comment (sambev) [reply index.js:8]: I was actually thinking about that. I'm not really sure what the behavior is but it's probably safest to add it?

Comment (whiteinge) [reply index.js:8]: I think so -- all our downstream projects will already use it (or want to use it).

**Review by whiteinge — APPROVED** [review]
:shipit:

**Pull request merged**